model/clusters_mgmt/v1/azure_etcd_encryption_type.model

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (c) 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// etcd data encryption settings.
+// Contains the necessary attributes to support etcd encryption for Azure based clusters.
+// Required during creation.
```

Comment:
This is not correct. At API level it is not required during creation. It will default to having data level encryption with platform managed keys.
Yes, that default behavior will not work initially. We need to separate what is the behavior that will be applied by default from what the backend supports or will return an error with.

Reply:
If we want we can add, on top of all the previously mentioned changes: currently only etcd data encryption is supported with customer managed keys. Creating a cluster with platform managed keys will result in a failure creating the cluster.

We don't really need this as this is not exposed to the end users for aro-hcp, and this should not be released once we are generally available but if we want to mention something like this and later remove it that's ok too.